appium/webdriver/extensions/gsm.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from selenium import webdriver
+from ..mobilecommand import MobileCommand as Command
+
+
+class Gsm(webdriver.Remote):
+
+    (
+        NONE_OR_UNKNOWN,
+        POOR,
+        MODERATE,
+        GOOD,
+        GREAT
+    ) = list(map(str, range(5)))  # TODO: Need to check where int 0 will be dropped as arg
```

Comment:
Here is log when int 0, not "0", is used.
4 is used instead of 0.

```
[HTTP] --> POST /wd/hub/session/806d31f9-84d0-4bd4-8264-cd6af8dd7ebb/appium/device/gsm_signal
(★) [HTTP] {"signalStrength":0,"sessionId":"806d31f9-84d0-4bd4-8264-cd6af8dd7ebb"}
[W3C (806d31f9)] Calling AppiumDriver.gsmSignal() with args: [null,"806d31f9-84d0-4bd4-8264-cd6af8dd7ebb"]
[ADB] gsm signal-profile  changes the reported strength on next (15s) update.
[ADB] Getting connected emulators
[ADB] Getting connected devices...
[ADB] 1 device(s) connected
[ADB] 1 emulator(s) connected
(★)[ADB] Running '/Users/atsushimori/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 emu gsm signal-profile 4'
```

Reply:
the 4 is the default value
https://github.com/appium/appium-adb/blob/527c8b417d7b62717ececfdf068309b0ee4dbd8f/lib/tools/adb-emu-commands.js#L165

Reply:
Yes, I've checked.
It's problem that 4(Gsm.GREAT) is set on the device when ```self.driver.set_gsm_signal(0)```(Gsm.NONE_OR_UNKNOWN) is called. But ```self.driver.set_gsm_signal('0')``` works.

I'll look into appium-adb codes.

Reply:
Your `[W3C (806d31f9)] Calling AppiumDriver.gsmSignal() with args: [null,"806d31f9-84d0-4bd4-8264-cd6af8dd7ebb"]`  should be like below when you send the request to server.
`null` is the cause of using `4` which is the default value.

```
[debug] [W3C (0df92b78)] Calling AppiumDriver.gsmSignal() with args: [0,"0df92b78-87ce-441e-93c7-2baf70eef29d"]
```

Reply:
Thanks. Hum, I'll keep current implementation and remove my comments.

string "0" works.

```
[HTTP] --> POST /wd/hub/session/604c1c76-bb99-4e47-b998-7ba2590a9fa6/appium/device/gsm_signal
(★)[HTTP] {"signalStrengh":"0","sessionId":"604c1c76-bb99-4e47-b998-7ba2590a9fa6"}
[W3C (604c1c76)] Calling AppiumDriver.gsmSignal() with args: ["0","604c1c76-bb99-4e47-b998-7ba2590a9fa6"]
[ADB] gsm signal-profile  changes the reported strength on next (15s) update.
[ADB] Getting connected emulators
[ADB] Getting connected devices...
[ADB] 1 device(s) connected
[ADB] 1 emulator(s) connected
(★)[ADB] Running '/Users/atsushimori/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 emu gsm signal-profile 0'
```

Reply:
I wonder Python client has an issue to filter `0`.
Could you add a unit test as the zero value?

Mine was by Ruby client. If JSON request has `0`, Appium should work.

Reply:
My guess is like below.
I'll look into codes more.

```
➜  mobile git:(master) ✗ python
Python 3.7.1 (default, Dec 29 2018, 23:35:51) 
>>> if 0:
...   print('hoge')
... 
>>> if 1:
...   print('hoge')
... 
hoge
```

Reply:
so far, https://github.com/appium/python-client/pull/357/files#r279606290 is needed. I ensured.
Then, `0` should work

Reply:
Thanks for your info, it works.